nav2_motion_primitives/src/stop.cpp

```diff
@@ -70,4 +70,9 @@ nav2_tasks::TaskStatus Stop::onCycleUpdate(nav2_tasks::StopResult & /*result*/)
   return nav2_tasks::TaskStatus::SUCCEEDED;
 }
 
+bool Stop::pathIsClear()
```

Comment:
This is a good instance where this method doesn't necessarily make sense. I think the API should just have an `IsValid()` method to return true if so or false when it couldnt do something necessary. It should be up to the primative to check if the path is clear because if it wasn't, it shouldnt be able to compute a valid trajectory/set of actions

Reply:
I think this requires more discussion. I'm assuming all motion primitives produce some robot movement and requires to check if the region of operation is clear. If that's the case, we should make it explicit on the API, that's why I have the `pathIsClear` method on the `MotionPrimitive` base class.

Now about the stop primitive, I don't think it should be a primitive but instead a method on the robot class.

Reply:
It's not that straight forward, it would require some changes to bt_navigator. Will follow up with #575.